src/components/DownloadsGraph.tsx

```diff
@@ -0,0 +1,70 @@
+import { LineChart } from '@mui/x-charts/LineChart';
+import { useState, useEffect } from 'react';
+import PackageRecommender from './PackageRecommender';
+import dayjs from 'dayjs';
+import numeral from 'numeral'; // Import numeral library
+import { Typography } from '@mui/material';
+import Stack from '@mui/material/Stack';
+
+type DateArray = Date[];
+type NumberArray = number[];
+
+interface PackageData {
+  collected: {
+    npm: {
+      downloads: {
+        from: string;
+        count: number;
+      }[];
+    };
+  };
+}
+
+export default function DownloadsGraph({ packageOne, packageTwo }: { packageOne: PackageData; packageTwo: PackageData; }) {
+  const [xDataPoints1, setXDataPoints1] = useState<DateArray>([]);
+  const [yDataPoints1, setYDataPoints1] = useState<NumberArray>([]);
+  const [xDataPoints2, setXDataPoints2] = useState<DateArray>([]);
+  const [yDataPoints2, setYDataPoints2] = useState<NumberArray>([]);
+
+  useEffect(() => {
+    function dateConversion(packageData: PackageData, setXDataPoints: (points: DateArray) => void, setYDataPoints: (points: NumberArray) => void) {
```

Comment:
Move this function outside useEffect.
The type for setter is not correct in the params.